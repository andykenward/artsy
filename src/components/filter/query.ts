import { DocumentNode } from 'graphql';
import gql from 'graphql-tag';

export const QUERY_FILTER: DocumentNode = gql`
  query FilterArtworks($color: String, $page: Int) {
    artworksConnection(color: $color, page: $page, first: 30, after: "") {
      pageInfo {
        hasNextPage
        endCursor
      }
      id
      edges {
        cursor
        node {
          id
          href
          title
          artistNames
          imageTitle
          image {
            placeholder
            resized(width: 400) {
              url
            }
          }
        }
      }
    }
  }
`;
