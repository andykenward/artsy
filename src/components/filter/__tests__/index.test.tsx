import { MockedResponse, wait } from '@apollo/react-testing';
import { act } from '@testing-library/react';
import React from 'react';
import { Filter } from '..';
import { useFilterArtworksQuery } from '../../../generated/graphql';
import { renderWithApp } from '../../../utils/tests';
import { QUERY_FILTER } from '../query';
import { FILTER_RESULT_DATA, FILTER_RESULT_DATA_EMPTY } from '../__fixtures__';

/**
 * FYI GraphQL Fragments do not currently work in Queries for testing
 */

describe('<Filter />', () => {
  const MOCKS: MockedResponse[] = [
    {
      request: {
        query: QUERY_FILTER,
        variables: {},
      },
      result: {
        data: FILTER_RESULT_DATA,
      },
    },
  ];

  it('should handle FilterArtworks query', async () => {
    const Component = () => {
      const { data, loading } = useFilterArtworksQuery();
      if (!loading) {
        expect(data).toEqual(FILTER_RESULT_DATA);
      }
      return null;
    };
    act(() => {
      renderWithApp(<Component />, MOCKS);
    });
  });

  it('should render loading state initially', async () => {
    await act(async () => {
      const root = renderWithApp(<Filter />, MOCKS);
      expect(root.container.firstChild).toMatchSnapshot();
    });
  });

  it('should render success response', async () => {
    await act(async () => {
      const root = renderWithApp(<Filter />, MOCKS);
      await wait(0); // wait for response
      expect(root.container.firstChild).toMatchSnapshot();
    });
  });

  it('should render empty success response', async () => {
    await act(async () => {
      const MOCKS: MockedResponse[] = [
        {
          request: {
            query: QUERY_FILTER,
            variables: {},
          },
          result: {
            data: FILTER_RESULT_DATA_EMPTY,
          },
        },
      ];
      const root = renderWithApp(<Filter />, MOCKS);
      await wait(0); // wait for response
      expect(root.container.firstChild).toMatchSnapshot();
      // TODO check for string in dom
    });
  });

  it.todo('error state');
});
