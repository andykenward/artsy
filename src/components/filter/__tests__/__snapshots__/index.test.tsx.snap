// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render empty success response 1`] = `
.c1 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 18px;
  line-height: 26px;
}

.c0 {
  background-color: #f4f4f4;
  padding: 1rem;
  opacity: 1;
  -webkit-transition: opacity 0.5s ease;
  transition: opacity 0.5s ease;
}

<div
  class="c0"
>
  <div
    class="c1 "
    font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
    font-size="18px"
  >
    No Artwork found with that colour
  </div>
</div>
`;

exports[`<Filter /> should render loading state initially 1`] = `
.c0 {
  background-color: #f4f4f4;
  padding: 1rem;
  opacity: 0.1;
  -webkit-transition: opacity 0.5s ease;
  transition: opacity 0.5s ease;
}

<div
  class="c0"
/>
`;

exports[`<Filter /> should render success response 1`] = `
.c7 {
  width: 100%;
  height: 100%;
  -webkit-transition: opacity 0.25s;
  transition: opacity 0.25s;
}

.c6 {
  background-color: #E5E5E5;
  -webkit-animation: hJdJWq 2s ease-in-out infinite;
  animation: hJdJWq 2s ease-in-out infinite;
}

.c8 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 0.75rem;
  line-height: 1rem;
  padding-top: 0.5rem;
}

.c9 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 0.75rem;
  line-height: 1rem;
}

.c5 {
  background-color: white;
  position: relative;
  width: 100%;
  overflow: hidden;
}

.c0 {
  background-color: #f4f4f4;
  padding: 1rem;
  opacity: 1;
  -webkit-transition: opacity 0.5s ease;
  transition: opacity 0.5s ease;
}

.c1 {
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
  -webkit-columns: 1 100%;
  columns: 1 100%;
  padding: 0;
  margin: 0;
  list-style: none;
  -webkit-columns: 2;
  columns: 2;
}

.c2 {
  display: inline-block;
  width: 100%;
  padding: 0.5rem 0;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:focus {
  outline: 0.3125rem auto rgba(0,103,244,0.247);
}

.c4 {
  padding: 0;
  margin: 0;
}

@media (min-width:56.25rem) {
  .c1 {
    -webkit-columns: 6 12.5rem;
    columns: 6 12.5rem;
  }
}

<div
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <a
        aria-label="Marlene"
        class="c3"
        href="/artwork/bruno-walpoth-marlene-1"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 152.74463007159903%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Bruno Walpoth, ‘Marlene’, 2016"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=610&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F1zY9HAtBJIbj5UDL-4u9ag%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Bruno Walpoth
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Marlene
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Endymion"
        class="c3"
        href="/artwork/louis-appleby-endymion-1"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 133.33333333333331%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Louis Appleby, ‘Endymion’, 2019"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=533&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2Fogu2T1JEBnzpwAyn2Jgziw%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Louis Appleby
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Endymion
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Vintage Exhibition Poster"
        class="c3"
        href="/artwork/lukman-glasgow-vintage-exhibition-poster"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 132.1705426356589%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Lukman Glasgow, ‘Vintage Exhibition Poster’, 1974"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=399&height=528&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FD6YZGkuGQ1be1ueHWiW2mg%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Lukman Glasgow
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Vintage Exhibition Poster
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Vintage Exhibition Poster"
        class="c3"
        href="/artwork/robert-harvey-vintage-exhibition-poster"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 146.484375%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Robert Harvey, ‘Vintage Exhibition Poster’, 1969/1970"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=585&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2Ftq63_4bkB2iiaabz7fJ6fg%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Robert Harvey
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Vintage Exhibition Poster
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Painter's Hand"
        class="c3"
        href="/artwork/tom-poelmans-painters-hand"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 72.52094028640907%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Tom Poelmans, ‘Painter's Hand’, 2019"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=290&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FTnsdDAU9G38R9OLAo9R5Pw%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Tom Poelmans
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Painter's Hand
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Untitled"
        class="c3"
        href="/artwork/guillermo-kuitca-untitled-44"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 22.65331664580726%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Guillermo Kuitca, ‘Untitled’, 2001"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=90&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FXNeaMm3MJG1xh8zJJRi3eQ%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Guillermo Kuitca
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Untitled
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Firebird"
        class="c3"
        href="/artwork/barbara-nanning-firebird"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 75%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Barbara Nanning, ‘Firebird’, 2018"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=300&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2Fu1hVHmBBp3eknM9A8bUKkQ%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Barbara Nanning
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Firebird
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="18-73"
        class="c3"
        href="/artwork/barbara-nanning-18-73"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 75%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Barbara Nanning, ‘18-73’, 2018"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=300&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FPMUVqcWLyxm_AcVYqUiVkg%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Barbara Nanning
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              18-73
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="17-46"
        class="c3"
        href="/artwork/barbara-nanning-17-46"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 75%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Barbara Nanning, ‘17-46’, 2019"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=300&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FwSouvhTmDKDmyGKQLGeeHw%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Barbara Nanning
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              17-46
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="18-82"
        class="c3"
        href="/artwork/barbara-nanning-18-82"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 75%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Barbara Nanning, ‘18-82’, 2019"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=300&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F0rp_TbaNy7uMIY3LAa_W8Q%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Barbara Nanning
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              18-82
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="18-84"
        class="c3"
        href="/artwork/barbara-nanning-18-84"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 75%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Barbara Nanning, ‘18-84’, 2018"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=300&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FksgMLxFHs3qR_PGI3AUnFg%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Barbara Nanning
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              18-84
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Ojai Festival 1980"
        class="c3"
        href="/artwork/richard-diebenkorn-ojai-festival-1980-1"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 174.42307692307693%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Richard Diebenkorn, ‘Ojai Festival 1980’, 1980"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=697&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FpgxynIUzaIHktXnHd491qg%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Richard Diebenkorn
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Ojai Festival 1980
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="17-32"
        class="c3"
        href="/artwork/barbara-nanning-17-32"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 75%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Barbara Nanning, ‘17-32’, 2019"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=300&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F7S_LY7ehO2skDt8p7Pp3WA%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Barbara Nanning
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              17-32
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="18-62"
        class="c3"
        href="/artwork/barbara-nanning-18-62"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 75%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Barbara Nanning, ‘18-62’, 2019"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=300&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FuK6Sk3wPAd1nJrbgo1R62A%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Barbara Nanning
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              18-62
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Lamborghini"
        class="c3"
        href="/artwork/kaxx-lamborghini"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 178.27298050139277%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="KAXX, ‘Lamborghini’, 2018"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=713&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FgVIfGaBnjIFW-poUvqgmBA%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                KAXX
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Lamborghini
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Mask 9"
        class="c3"
        href="/artwork/ryan-bock-mask-9"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 139.93362831858406%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Ryan Bock, ‘Mask 9’, 2018"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=559&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F78YENpVkLDNOv_Hje7ugJg%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Ryan Bock
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Mask 9
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Justice"
        class="c3"
        href="/artwork/kilroy-savage-justice"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 125.73099415204678%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Kilroy Savage, ‘Justice’, 2019"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=502&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FcLyedwd3zCUYvxlNJVuwsg%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Kilroy Savage
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Justice
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Alley Cat"
        class="c3"
        href="/artwork/alexis-kandra-alley-cat"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 101.86757215619696%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Alexis Kandra, ‘Alley Cat’, 2019"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=407&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FaCtYVJPYHxCkpRZsAEsylw%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Alexis Kandra
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Alley Cat
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="New Beginning"
        class="c3"
        href="/artwork/ekaterina-popova-new-beginning"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 132.6963906581741%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Ekaterina Popova, ‘New Beginning’, 2019"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=530&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FEmXbo80hXQzZk37JDo5JQw%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Ekaterina Popova
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              New Beginning
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Milk 1"
        class="c3"
        href="/artwork/alex-tea-milk-1"
      >
        <figure
          class="c4"
        >
          <div
            class="c5"
            style="padding-bottom: 106.94121554300897%;"
          >
            <div
              class="sc-AxjAm c6"
            >
              <img
                alt="Alex Tea, ‘Milk 1’, 2018"
                class="c7"
                height="100%"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=427&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FJ206kpxw9BxzUVb5cPYt-w%2Flarge.jpg"
                style="position: absolute; object-fit: contain; width: 100%; height: 100%; opacity: 0;"
                width="100%"
              />
              <noscript />
            </div>
          </div>
          <figcaption>
            <div
              class="c8 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              <strong>
                Alex Tea
              </strong>
            </div>
            <div
              class="c9 "
              font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
              font-size="0.75rem"
            >
              Milk 1
            </div>
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</div>
`;
